### PR TITLE
Refresh flow state on query update

### PR DESCRIPTION
**Pull request by alexandrudanpop**
Fixes #<issue_number>.

## Testing Checklist

Check all that apply:

- [ ] I tested the feature thoroughly, including edge cases

- [ ] I verified all affected areas still work as expected

- [ ] Automated tests were added/updated if necessary

- [ ] Changes are backwards compatible with any existing data, otherwise a migration script is provided

## Visual Changes (if applicable)

If there are UI/UX changes, include at least one of the following:

- Screenshots
- Loom video
- Preview deployment link

### CODE REVIEW

File: packages/react-ui/src/app/features/builder/builder-state-provider.tsx [hunk 1]
```diff
@@ -18,6 +18,15 @@ export function BuilderStateProvider({
   const storeRef = useRef<BuilderStore>();
   const { checkAccess } = useAuthorization();
 
+  useEffect(() => {
+    if (storeRef.current && props.flow && props.flowVersion) {
+      storeRef.current.setState(() => ({
```
Comment (cezudas) [inline packages/react-ui/src/app/features/builder/builder-state-provider.tsx]: Isn't passing just the partial object state to useState (not a function that returns the object) enough?

```
 storeRef.current.setState({
        flow: props.flow,
        flowVersion: props.flowVersion,
      });
```

Comment (alexandrudanpop) [reply packages/react-ui/src/app/features/builder/builder-state-provider.tsx]: done
